### PR TITLE
[21] Feature : Import user contacts

### PR DESCRIPTION
**Pull request by loicleclere**
This feature will be completed in a future release, so that it can be implemented in the follower to suggest close friends.

### CODE REVIEW

File: backend/microservices/UserAccount/src/controllers/UserControllers.ts [hunk 1]
```diff
@@ -313,6 +315,31 @@ class UserControllers {
...
+    const { contacts } = await ctx.req.json()
+
+    const allUsers = await prisma.user.findMany()
+
+    const contactsWithPhoneNumbers = contacts.filter(
+      (contact: Contact) => contact.phoneNumbers && contact.phoneNumbers[0]
+    )
+
+    const friendsPromises: Promise<User>[] = []
+    if (contactsWithPhoneNumbers.length > 0) {
```
Comment (zrichard22) [inline backend/microservices/UserAccount/src/controllers/UserControllers.ts:329]: Do a early return when no contact have a phone number

File: mobile-app/app/(auth)/(tabs)/index.tsx [hunk 2]
```diff
@@ -1,20 +1,58 @@
...
-          title='Sign Out'
           onPress={() => {
             signOut()
           }}
-        />
+        >
+          Sign Out
+        </Button>
+        <Text color={'#000'} textAlign='center' fontSize={16}>
+          Vous avez {friendsContacts ? friendsContacts.length : 0} amis
```
Comment (Lucas-Senoville) [inline mobile-app/app/(auth)/(tabs)/index.tsx:53]: Please change it to english by default

**Review by Lucas-Senoville — CHANGES_REQUESTED** [review]

File: mobile-app/app/(auth)/(tabs)/index.tsx [hunk 3]
```diff
@@ -1,20 +1,58 @@
...
 
 export default function Tab() {
-  const { signOut } = useSession()
+  const { signOut, isLoading, session } = useSession()
+  const [friendsContacts, setFriendsContacts] = useState<User[] | null>(null)
+
+  useEffect(() => {
+    const fetchContacts = async () => {
+      const contacts = await getContacts()
+      if (contacts) {
```
Comment (zrichard22) [inline mobile-app/app/(auth)/(tabs)/index.tsx:18]: Multiple nested if statement

File: mobile-app/utils/getStorageAvailable.ts [hunk 4]
```diff
@@ -1,4 +1,8 @@
-const getStorageAvailable = (isLoading: boolean, storage: string | null) =>
-  !isLoading && storage
+import { Session } from '@/types/session.type'
+
+const getStorageAvailable = (
+  isLoading: boolean,
+  storage?: string | null | Session
```
Comment (zrichard22) [inline mobile-app/utils/getStorageAvailable.ts:5]: Objects before primitive before null/undefined

**Review by zrichard22 — DISMISSED** [review]

**Review by zrichard22 — APPROVED** [review]

**Review by Lucas-Senoville — APPROVED** [review]

**Pull request merged**